src/hotspot/share/logging/logFileStreamOutput.cpp

```diff
@@ -86,19 +87,49 @@ class FileLocker : public StackObj {
   }
 };
 
+static bool write_error_is_shown = false;
+
+bool LogFileStreamOutput::flush() {
+  bool result = true;
+  if (fflush(_stream) != 0) {
+    if (!write_error_is_shown) {
+      jio_fprintf(defaultStream::error_stream(),
+                  "Could not flush log: %s (%s (%d))\n", name(), os::strerror(errno), errno);
+      jio_fprintf(_stream, "\nERROR: Could not flush log (%d)\n", errno);
+      write_error_is_shown = true;
+    }
+    result = false;
+  }
+  return result;
+}
+
+#define WRITE_LOG_WITH_RESULT_CHECK(op, total)                \
+{                                                             \
+  int result = op;                                            \
+  if (result <= 0) {                                          \
```

Comment:
May be not completely correct for all values of op:
- fprintf returns a negative value (0 is valid if length to write was 0). I did not check jio_fprintf but I assume the return semantics are the same?
- write_decorations returns 0 if it did not write any decorations, which is valid

Reply:
`jio_fprintf()` would call `vfprintf()`, so I concerned in case that `vfprintf()` cannot write enough. Callers of `jio_fprintf()` other than `write_decorations()` would write 1 byte at least.
However `write_decorations()` might not write any chars, and also `vfprintf()` in glibc does not seem to return halfway value. So I updated condition to `< 0`.